src/storage/storageController.ts

```diff
@@ -90,24 +90,14 @@ export default class StorageController {
   }
 
   getUserIdentity() {
-    const userId = this.get(StorageVariables.GLOBAL_USER_ID);
     let anonymousId = this.get(StorageVariables.GLOBAL_ANONYMOUS_ID);
 
+    // The anonymousId becomes required with analytics-node v6.
     if (!anonymousId) {
       anonymousId = uuidv4();
       void this.update(StorageVariables.GLOBAL_ANONYMOUS_ID, anonymousId);
     }
 
-    // Initially, we used `userId` as Segment user identifier, but this usage is being deprecated.
-    // The `anonymousId` should be used instead.
-    // We keep sending `userId` to Segment for old users though to preserve their analytics.
```

Comment:
That's a good point, but the transition to anonymousId was done 2 years ago, is safe to just drop the userId by now and approved by the analytics team: https://github.com/mongodb-js/vscode/pull/394

Reply:
I think this note made sense when anonymousId was first introduced, to have some consistency. But now that anonymousId has been in use for about a year, this could only be disruptive for really old analytics or users who haven't updated for a year. I'm hoping we don't need to support these edge cases.
The request was to use userId only for AUID, and for that we need to let go of these legacy anonymous userIds. https://jira.mongodb.org/browse/COMPASS-7610